Week-4/callApplyBind/script.js

```diff
@@ -0,0 +1,64 @@
+// Calculator class with basic arithmetic methods
+class Calculator {
+    add(a, b) {
+      return a + b;
+    }
+  
+    subtract(a, b) {
+      return a - b;
+    }
+  
+    multiply(a, b) {
+      return a * b;
+    }
+  
+    divide(a, b) {
+      if (b === 0) {
+        throw new Error('Cannot divide by zero.');
+      }
+      return a / b;
+    }
+  }
+  
+  // ScientificCalculator class that extends Calculator with advanced methods
+  class ScientificCalculator extends Calculator {
```

Comment:
Consider checking for data types before performing operations in the Calculator methods to ensure robustness. For example, ensure that a and b are numbers.